runtime/hashing/common/src/aws/smithy/kotlin/runtime/hashing/HashingSink.kt

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.hashing
+
+import aws.smithy.kotlin.runtime.io.SdkSink
+import aws.smithy.kotlin.runtime.io.internal.SdkSinkObserver
+
+/**
+ * A sink which hashes data as it is being written to
+ * @param hash The [HashFunction] to hash data with
+ * @param sink the [SdkSink] to hash
+ */
+public class HashingSink(public val hash: HashFunction, sink: SdkSink) : HashFunction by hash, SdkSinkObserver(sink) {
```

Comment:
**Correctness**: Why implement `HashFunction`? Is there a use case for calling `update`/`reset` on a sink? Will we ever pass a `HashingSink` as an argument to `ByteArray.hash`?

Reply:
there's no particular reason, and you're right, there are no strong use cases for directly calling update / reset on this sink, so I'll remove this implementation